Monika-After-Story/MonikaModDev
# Follow-up for players who quit smoking

EDIT:
Code has been cleaned up and requires testing:

## Testing:
- Verify that the update script works (calling the update label with `persistent._mas_do_smoke_quit` being `True` will set up the start-date for the `monika_smoking_quit` ev.
- Verify that lying to Monika twice about having smoked results in the main smoking topic being locked
- Verify that in flows where you tell Monika you're trying to quit, the follow-up topic will queue (or requeue) with a start-date of a month in the future.


Original post below
-------------------------------------------------------------------------------------
Add the new event monika_smoking_quit and expands upon monika_smoking.

These lines (thanks @multimokia ) should go in `updates.rpy` under the appropriate milestone (I don't know which it is) :
```
if persistent._mas_pm_do_smoke_quit:
    smoking_ev = mas_getEV('monika_smoking')
    if smoking_ev and smoking_ev.last_seen:
        mas_getEV("monika_smoking_quit").mas_utils.add_months(smoking_ev.last_seen, 1)
```
